cli: Fix compilation without LANG/LC_ALL set

zarybnicky opened the pull request:
Embedded files are now assumed to be UTF-8 independently of the system locale. This also replaces the file-embed dependency with `Hledger.Cli.TH`, where `embedStringFile` has been changed to assume UTF-8 (`utf8_bom`).

Compilation without a locale (e.g. in `nix-shell --pure` or `docker run -it fedora:28 bash` as @talex5 mentioned) now works without a problem.

~~I've also changed the `nix:` sections of `stack.yaml`s to use pure nix shells, and added the required dependencies of happy/alex - `stack build --nix` works for all build configs. (I'm on NixOS, so I can't actually develop without `--nix`, and compiling in a pure shell is also a nice bonus.)~~

Fixes #420, #813. ~~This is also a better solution to the closed #669.~~

simonmichael commented:
@zarybnicky, thanks for making the build more robust. Could you perhaps:

- submit embedStringFileUtf8 upstream - it seems like a useful feature. 
- include the notes above in the commit description(s)
- separate the nix/stack updates into their own commit (or, mention why they belong with this one)

zarybnicky commented:
I've opened an upstream issue, snoyberg/file-embed#27. I've added to the commit description, and removed the Nix-related changes (I might open another pull request later, it's nothing urgent, just a nice-to-have)

simonmichael commented:
Thanks @zarybnicky. I guess we should do a shorter workaround with less code duplication following the suggestions on that issue.

zarybnicky commented:
Oops, didn't mean to click that.

simonmichael commented:
Great, I guess this fixes [this](https://github.com/simonmichael/hledger/issues/420#issuecomment-406971517) and [this](https://github.com/simonmichael/hledger/issues/813) and is ready for merge ?

zarybnicky commented:
Yup, I think so.

simonmichael commented:
Thanks!
